feat: easier phantomjs debugging

marionebl opened the pull request:
This PR aims at making phantomjs debugging easier:

* add `--phantom-remote-debugger-port=[port]` which maps to phantoms `--remote-debugger-port=[port]`
* add `--phantom-remote-debugger-autorun` which maps to phantoms `--remote-debugger-autorun`
* add `phantom.onError` handler
* process messages from `phantom.onError`, `page.onError` as `exceptions`, exit the phantom run with `1` if an exception occurs

vvo commented on lib/PhantomBrowser.js:
```diff
@@ -26,7 +26,7 @@ PhantomBrowser.prototype.start = function() {
 
     var phantomjs = require('phantomjs');
-    var binpath = phantomjs.path;
+    var binpath = phantomjs.path || '/Users/marneb/Projekte/labs/jogwheel/node_modules/.bin/phantomjs';
```
woops

vvo commented:
Can you show how to use the debugger and what does using phantom.onError provides comparing to now?

marionebl commented:
Hey @vvo,

removed the debugging leftover and squashed the commits.

## CLI Usage
```
# Start phantom runner waiting for remote debugging session
zuul test.js --phantom --phantom-remote-debugger-port=1337
# Open phantom remote debugger session in browser
open http://localhost:1337

# Start phantom runnner and run tests right await but allow connection of remote debugger
zuul test.js --phantom --phantom-remote-debugger-port=1337 --phantom-remote-debugger-autorun
```

## New error handler
`phantom.onError` is intended as last safety net if something goes wrong before `page.onError` kicks in. I followed the explicit recommendation of the [phantomjs docs](http://phantomjs.org/api/phantom/handler/on-error.html):

> This is the closest it gets to having a global error handler in PhantomJS, and so it is a best practice to set this onError handler up in order to catch any unexpected problems.

vvo commented:
Awesome, can you update the wiki to add the documentation? https://github.com/defunctzombie/zuul/wiki/Phantom-js

marionebl commented:
Updated https://github.com/defunctzombie/zuul/wiki/Phantom-js.
:+1: for releasing!

vvo commented:
thanks!
